lib/lib/interfaces.d.ts

```diff
@@ -2,7 +2,81 @@ export interface Response {
     error: boolean;
     data: any;
 }
+export interface GenerateMnemonicRequest {
+    entropy?: 128 | 160 | 192 | 224 | 256;
+    length?: 12 | 15 | 18 | 21 | 24;
+}
+export interface GenSeedRequest {
+    password?: string;
+}
+export interface GenerateExtendedKeyRequest {
+    network?: string;
+    mnemonic: string;
+    password?: string;
+}
+export interface GenerateExtendedKeyResponse {
+    fingerprint: string;
+    mnemonic: string;
+    xprv: string;
+}
+export declare type WPKH = 'default' | null | '' | 'p2wpkh' | 'wpkh';
+export declare type P2PKH = 'p2pkh' | 'pkh';
+export declare type SHP2WPKH = 'shp2wpkh' | 'p2shp2wpkh';
+export interface CreateDescriptorRequest {
+    type: WPKH | P2PKH | SHP2WPKH | 'multi';
+    /**
+     * Set useMnemonic: true, if want to create desciptor using mnemonic* and password*.
+     * Set useMnemonic: false, if want to create descriptor using xprv*
+     */
+    useMnemonic: boolean;
+    /**
+     * Required if useMnemonic: false
+     */
+    xprv?: string;
+    /**
+     * Required if useMnemonic: true
+     */
+    mnemonic?: string;
+    /**
+     * Required if useMnemonic: true
+     */
+    password?: string;
+    /**
+     * Required if useMnemonic: true
+     */
+    network?: 'bitcoin' | 'testnet' | 'signet' | 'regtest';
+    /**
+     * If want to use custom path instead of default(/84'/1'/0'/0/*)
+     */
+    path?: string;
+    /**
+     * required if type is multi
+     * can't be 0 or grator than number public keys
+     * */
+    thresold?: number;
+    /**
+     * Array of public keys
+     */
+    publicKeys?: Array<string>;
+}
+export interface createWalletRequest {
+    mnemonic?: string;
+    descriptor?: string;
+    useDescriptor?: boolean;
```

Comment:
A simpler API could be to use `descriptor` is set, otherwise, use `mnemonic`. Throw if none or both are defined.

Reply:
Yes, added an issue for it 😊